source/uwp/Renderer/lib/Util.cpp

```diff
@@ -161,677 +132,431 @@ try
-                                  _Outptr_ HSTRING* resultFontFamilyName) noexcept
-try
+winrt::hstring GetFontFamilyFromFontType(winrt::AdaptiveHostConfig const& hostConfig, winrt::FontType const& fontType)
 {
-    HString result;
-    ABI::AdaptiveCards::Rendering::Uwp::IAdaptiveFontTypeDefinition* typeDefinition;
-
-    // get FontFamily from desired style
-    RETURN_IF_FAILED(GetFontType(hostConfig, fontType, &typeDefinition));
-    RETURN_IF_FAILED(typeDefinition->get_FontFamily(result.GetAddressOf()));
-    if (result == NULL)
+    // Get FontFamily from desired style
+    auto typeDefinition = GetFontType(hostConfig, fontType);
+    auto fontFamily = typeDefinition.FontFamily();
+    if (fontFamily.empty())
     {
-        if (fontType == ABI::AdaptiveCards::ObjectModel::Uwp::FontType::Monospace)
+        if (fontType == winrt::FontType::Monospace)
         {
             // fallback to system default monospace FontFamily
-            RETURN_IF_FAILED(UTF8ToHString("Courier New", result.GetAddressOf()));
+            fontFamily = UTF8ToHString("Courier New");
         }
         else
         {
             // fallback to deprecated FontFamily
-            RETURN_IF_FAILED(hostConfig->get_FontFamily(result.GetAddressOf()));
-            if (result == NULL)
+            fontFamily = hostConfig.FontFamily();
+            if (fontFamily.empty())
             {
                 // fallback to system default FontFamily
-                RETURN_IF_FAILED(UTF8ToHString("Segoe UI", result.GetAddressOf()));
+                fontFamily = UTF8ToHString("Segoe UI");
             }
         }
     }
-    return result.CopyTo(resultFontFamilyName);
+    return fontFamily;
 }
-CATCH_RETURN;
 
-HRESULT GetFontSizeFromFontType(_In_ ABI::AdaptiveCards::Rendering::Uwp::IAdaptiveHostConfig* hostConfig,
-                                ABI::AdaptiveCards::ObjectModel::Uwp::FontType fontType,
-                                ABI::AdaptiveCards::ObjectModel::Uwp::TextSize desiredSize,
-                                _Out_ UINT32* resultSize) noexcept
-try
+uint32_t GetFontSizeFromFontType(winrt::AdaptiveHostConfig const& hostConfig, winrt::FontType const& fontType, winrt::TextSize const& desiredSize)
 {
-    UINT32 result;
-    ABI::AdaptiveCards::Rendering::Uwp::IAdaptiveFontTypeDefinition* fontTypeDefinition;
-    ABI::AdaptiveCards::Rendering::Uwp::IAdaptiveFontSizesConfig* sizesConfig;
-
-    // get FontSize from desired style
-    RETURN_IF_FAILED(GetFontType(hostConfig, fontType, &fontTypeDefinition));
-    RETURN_IF_FAILED(fontTypeDefinition->get_FontSizes(&sizesConfig));
-    RETURN_IF_FAILED(GetFontSize(sizesConfig, desiredSize, &result));
+    winrt::AdaptiveFontTypeDefinition fontTypeDefinition = GetFontType(hostConfig, fontType);
+    winrt::AdaptiveFontSizesConfig sizesConfig = fontTypeDefinition.FontSizes();
+    uint32_t result = GetFontSize(sizesConfig, desiredSize);
 
+    // TODO: can we still use MAXUINT32?
```

Comment:
@paulcam206 could you please have a look at this?

Reply:
I believe `std::numeric_limits<unsigned int>::max()` is the more modern way we've done this in the shared model...